File: code-example-tests/javascript/driver/tests/time-series/auto-removal.test.js
import Expect from '../../utils/Expect.js';
import {
  updateCollectionOptions,
  getCollectionInfo,
  removeRemoval,
  cleanup,
} from '../../examples/time-series/auto-removal.js';

describe('Auto removal time series collection tests', () => {
  afterEach(async () => {
    await cleanup();
  });

  it('Should update collection options successfully', async () => {
    const result = await updateCollectionOptions();

    Expect.that(result)
      .shouldResemble({ ok: 1 })
      .withSchema({
        count: 1,
        requiredFields: ['ok'],
        fieldValues: { ok: 1 },
      });
  });

  it('Should return the correct collection expiry', async () => {
    const result = await getCollectionInfo();
    Expect.that(result).shouldMatch(86400);
  });

  it('Should remove expireAfterSeconds successfully', async () => {
    const result = await removeRemoval();
    Expect.that(result).shouldMatch(true);
  });
});

File: code-example-tests/csharp/driver/Tests/TimeSeries/AutoRemovalTest.cs
using DotNetEnv;
using Examples.TimeSeries;
using MongoDB.Bson;
using MongoDB.Driver;
using Utilities.Comparison;

namespace Tests.TimeSeries;

public class AutoRemovalTest
{
    private static readonly string Uri = DotNetEnv.Env.GetString("CONNECTION_STRING",
        "Env variable not found. Verify you have a .env file with a valid connection string.");

    [Test]
    public async Task TestCollectionOptionsUpdate()
    {
        var expected = new[]
        {
            new Dictionary<string, object> { { "ok", 1 } }
        };

        var updateCommandResult = await AutoRemoval.UpdateCollectionOptions();

        Expect.That(updateCommandResult)
            .ShouldResemble(expected)
            .WithSchema(new SchemaValidationOptions
            {
                Count = 1,
                RequiredFields = new[] { "ok" },
                FieldValues =
                    new Dictionary<string, object?>() { { "ok", 1 } }
            });
    }

    [Test]
    public async Task TestCollectionExpiry()
    {
        var result = await AutoRemoval.GetCollectionInfo();
        Expect.That(result).ShouldMatch(86400);
    }

    [Test]
    public async Task TestRemoveRemoval()
    {
        var result = await AutoRemoval.RemoveRemoval();
        Expect.That(result).ShouldMatch(true);
    }

    [TearDown]
    public void TearDown()
    {
        AutoRemoval.Cleanup();
    }
}